agent/utils/ticker.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"context"
+	"math/rand"
+	"time"
+)
+
+// JitteredTicker works like a time.Ticker except with randomly distributed ticks
+// between start and end duration.
+func JitteredTicker(ctx context.Context, start, end time.Duration) <-chan time.Time {
+	ticker := make(chan time.Time, 1)
+
+	go func() {
+		defer close(ticker)
+
+	loop:
+		for {
+			select {
+			case <-ctx.Done():
+				break loop
+			default:
+				time.Sleep(randomDuration(start, end))
+				sendNow(ticker)
+			}
+		}
+	}()
+
+	return ticker
+}
+
+func randomDuration(start, end time.Duration) time.Duration {
+	return time.Duration(start.Nanoseconds()+rand.Int63n(end.Nanoseconds()-start.Nanoseconds())) * time.Nanosecond
```

Comment:
Why `Nanoseconds` instead of `Milliseconds`? I don't think Nanoseconds is enough for the jitter, as it means that there could still be up to 1e9 requests within one second.